### DISCUSSION TITLE
Question about using Links with styled and functional components

### DISCUSSION THREAD

**Discussion by ben-me** [neuefische/questions#152] [Web] [answered]
Hello there,

in my app I'm using nextJS/link for client-side transitions. Usually Link is used in combination with an anchor tag but I want to use it with buttons.

The next documentation has something about custom components 

> [If the child is a custom component that wraps an <a> tag](https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag)
> If the child of Link is a custom component that wraps an <a> tag, you must add passHref to Link. This is necessary if you’re using libraries like [styled-components](https://styled-components.com/). Without this, the <a> tag will not have the href attribute, which hurts your site's accessibility and might affect SEO. If you're using [ESLint](https://nextjs.org/docs/basic-features/eslint#eslint-plugin), there is a built-in rule next/link-passhref to ensure correct usage of passHref.

It also features a part about functional components 

>[If the child is a functional component](https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-functional-component)
>If the child of Link is a functional component, in addition to using passHref, you must wrap the component in [React.forwardRef](https://reactjs.org/docs/react-api.html#reactforwardref)

My first idea was to use the link on a (styled) button like this: 
```jsx
<Link href="/abc">
   <Button>Got to page abc</Button>
</Link>
```

Do I now have to pass down the href to the styled component?

Or should I also include anchor tags? And do I need the passHref?

```jsx
<Link href="/rooms">
    <a>
       <Button>Show all rooms</Button>
    </a>
</Link>
```

There is also this information from the NextJS learn resource. 
>Note: Before Next.js 12.2, it was required that the Link component wrapped an <a> tag, but [this is nor required in versions 12.2 and above](https://nextjs.org/blog/next-12-2#:~:text=next/link%20no%20longer%20requires%20manually%20adding%20%3Ca%3E%20as%20a%20child.%20You%20can%20now%20opt%20into%20this%20behavior%20in%20a%20backward%2Dcompatible%20way.).

With this in mind I don't seem to need the anchor tag.

I am confused as to what is the right way to use the nextjs/link on html tags like buttons or styled components.

**Comment by NikRoe** [comment, marked as answer]
Hey Benedikt,

I would advise working with an `a` tag and styling the tag like a button.
This ensures accessibility of your links while visually appearing like a button and working like a link.

**Comment by ben-me** [comment]
Hey,

thanks for answering. Could you please explain further?
